Always use Unix-style line endings in generator output

nbes4 opened the pull request:
This PR adds:
- A centralized `serialize()` implementation.
 I think this one made sense since it was the same for `Device`, `Symbol`, `Package` and `Component`. If these implementations differ in the future we could remove it again, but for now it prevents some code duplication.  (I wasn't sure how to name it and where to put it, I opted for `serialize_common` and put it in the _outer_ `common.py` file, suggestions welcome)
- Always use Unix-style line endings (LF) when serializing entities (should fix development on Windows, see discussion [here](https://github.com/LibrePCB-Libraries/JST.lplib/pull/6))
- Tests

I tried the `librepcb-cli open-library --all --strict --check --minify-step .` command before the changes in this PR and it gave me the "Non-canonical file". After applying the changes in this PR it seemed to work just fine! :-)

Edit: I verified the Tests on my Windows machine. They failed before applying the changes, but passed after applying the changes from this PR.

dbrgn commented on common.py:
```diff
@@ -6,6 +6,7 @@
 import os.path
 import re
 from datetime import datetime
+from os import makedirs, path
```
The `os.path` import was already there... Could you replace usages of `os.path` with `path` instead, and remove the `o.spath` import?

dbrgn commented on entities/component.py:
```diff
@@ -212,11 +212,8 @@ def add_variant(self, variant: Variant) -> None:
-        with open(path.join(dir_path, 'component.lp'), 'w') as f:
-            f.write(str(self))
-            f.write('\n')
+        serialize_common(serializable=self,
+                         output_directory=output_directory,
+                         uuid=self.uuid,
+                         long_type='component',
+                         short_type='cmp')
```
Nit: Could you format the function calls like this instead?

```
serialize_common(
    serializable=self,
    output_directory=output_directory,
    uuid=self.uuid,
    long_type='component',
    short_type='cmp',
)
```

The reason is that if we ever rename the `serialize_common` function, we don't need to change every line in the function call just to re-align the arguments vertically...

rnestler replied:
See https://github.com/LibrePCB/librepcb-parts-generator/pull/133 for a suggestion to use ruff for formatting.

dbrgn reviewed (changes requested):
Looks good to me! Only small nits from my side.

Did you test whether the test actually fails when running on Windows without the fix?

nbes4 commented:
@dbrgn Thanks for the review, I implemented your feedback! 

> Did you test whether the test actually fails when running on Windows without the fix?

Yes, I forgot to mention that but I did run the tests on my Windows machine before applying the changes in this PR and they failed, but passed once I applied the changes 👍 I updated the initial PR description to make it more clear upfront 😄

rnestler reviewed (approved):
Maybe we should even add Windows to the testing matrix

nbes4 commented:
Thanks for merging 🎉 

@rnestler more coverage never hurts IMO 👍
